Initial sample (#1)
Azure-Samples/ms-identity-b2c-javascript-signin: README.md

---
page_type: sample
languages:
  - javascript
products:
  - azure-active-directory-b2c
  - microsoft-identity-platform
  - microsoft-authentication-library 
urlFragment: ms-identity-b2c-javascript-signin
description: "demonstrates a Vanilla JavaScript single-page application that lets users to sign-in to Azure AD B2C"
---

<div id="navbar">
    <p>In this tutorial: authentication, B2C, identity providers</p>
    <a href="/">Previous Tutorial</a>
    <a href="/">Next Tutorial</a>
    <a href="/">Contents</a>
</div>
<br>

# Vanilla JavaScript Single-page Application

 1. [Overview](#overview)
 1. [Scenario](#scenario)
 1. [Contents](#contents)
 1. [Prerequisites](#prerequisites)
 1. [Setup](#setup)
 1. [Registration](#registration)
 1. [Running the sample](#running-the-sample)
 1. [Explore the sample](#explore-the-sample)
 1. [About the code](#about-the-code)
 1. [Deployment](#deployment)
 1. [More information](#more-information)
 1. [Community Help and Support](#community-help-and-support)
 1. [Contributing](#contributing)
 1. [Code of Conduct](#code-of-conduct)

## Overview

> This sample demonstrates a Vanilla JavaScript single-page application that lets users to sign-in to Azure AD B2C. In doing so, it also illustrates various authentication and B2C concepts, such as **ID Tokens**, **OIDC** scopes, **social identities**, **identity providers**, **single-sign on** (SSO), **account selection**, **silent requests** and etc.

Reviewer: ```suggestion
This sample demonstrates a Vanilla JavaScript single-page application that can sign in users to Azure AD B2C. In doing so, it also illustrates various authentication and B2C concepts, such as **ID Tokens**, **OIDC** scopes, **social identities**, **identity providers**, **single-sign on** (SSO), **account selection**, **silent requests** and etc.
```

`sign-in` is when talking about a noun (e.g. `in the app logs, see my sign-in`)
`sign in` when talking about an action/verb (e.g. `see me sign in`)

![Overview](./ReadmeFiles/topology.png)

## Scenario

1. The client application uses the **MSAL.js** to obtain an **ID Token** from Azure AD B2C.
2. The **Id Token** is used to authenticate the user into their organization's B2C tenant.

## Contents

| File/folder           | Description                                |
|-----------------------|--------------------------------------------|
| `App/`                | Contains sample source code.                |

> | `ReadmeFiles/`        | Contains illustrations and etc.             |
> | `server.js`           | The license for the sample.                |

Reviewer: ```suggestion
| `ReadmeFiles/`        | Contains illustrations for the readme             |
| `server.js`           | The sample server code              |
```

| `CONTRIBUTING.md`     | Guidelines for contributing to the sample. |
| `LICENSE`             | The license for the sample.                |

## Prerequisites

- [Node.js](https://nodejs.org/en/download/) must be installed to run this sample.
- A modern web browser. This sample uses **ES6** conventions and will not run on **Internet Explorer**.
- [Visual Studio Code](https://code.visualstudio.com/download) is recommended for running and editing this sample.
- [VS Code Azure Tools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-node-azure-pack) extension is recommended for interacting with Azure through VS Code Interface.
- An Azure Active Directory (Azure AD) tenant. For more information on how to get an Azure AD tenant, see: [How to get an Azure AD tenant](https://azure.microsoft.com/documentation/articles/active-directory-howto-tenant/)
- A user account in your Azure AD tenant.

## Setup

### Step 1: Clone or download this repository

From your shell or command line:

```console
git clone https://github.com/Azure-Samples/ms-identity-b2c-javascript-signin.git
```

or download and extract the repository .zip file.

> > :warning: Given that the name of the sample is quiet long, and so are the names of the referenced packages, you might want to clone it in a folder close to the root of your hard drive, to avoid maximum file path length limitations on Windows.

Reviewer: ```suggestion
> :warning: Given that the name of the sample is quite long, and so are the names of the referenced packages, you might want to clone it in a folder close to the root of your hard drive, to avoid maximum file path length limitations on Windows.
```
